### PR TITLE
Add CUDA 11.1 migrator

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Blocked by https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/131 and https://github.com/conda-forge/docker-images/pull/160

### CODE REVIEW

**Comment by jaimergp** [comment]
That doesn't work even if tests are successful. CUDA 11 is back to 2.12 sysroots.

**Comment by jaimergp** [comment]
Ok, using `cdt_name` does work. I guess that's more sustainable in the future (and makes more sense since 2.17 is tied to cos7?). If this is not ok, we need to figure out a clean way to state that CUDA 11.0+ requires glibc 2.17+...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,22 +20,22 @@ outputs:
       run_exports:
         strong:
           - cudatoolkit {{ cuda_compiler_version }}|{{ cuda_compiler_version }}.*
-          - __glibc >=2.17  # [linux and cuda_compiler_version == "11.0"]
+          - __glibc >=2.17  # [linux and cuda_compiler_version in ("11.0", "11.1")]
```
Comment (kkraus14) [inline recipe/meta.yaml:23]: Can we use string comparisons so we don't have to keep adding here whenever a new CUDA version is released?
```suggestion
          - __glibc >=2.17  # [linux and cuda_compiler_version >= "11.0"]
```

Comment (jaimergp) [reply recipe/meta.yaml:23]: I tried with int-tuples but the selector syntax does not work well with them. See previous commits and maybe you'll see something I missed?

Comment (jaimergp) [reply recipe/meta.yaml]: Hm, I figured out a way to do it but it's _not_ pretty. See commit 7ca2d6a.

Comment (kkraus14) [reply recipe/meta.yaml]: Do we need to convert the major version to an int? I was suggesting using string comparison, but that does fail if/when there's a CUDA version 100 😄.

Comment (jaimergp) [reply recipe/meta.yaml]: 9.2 would be greater than 11 if we use string comparison:

```
>>> "9.2" > "11.0"
True
```

Comment (kkraus14) [reply recipe/meta.yaml]: Resolving this, forgot we're still supporting 9.2 for now.

**Comment by jaimergp** [comment]
I'll probably get back to this soon. Looks like all the missing pieces are available now!

**Pull request closed without merging**